docs/flux.rst

```diff
@@ -0,0 +1,212 @@
+.. _flux-page:
+
+**************
+Flux Framework
+**************
+
+`Flux Framework <https://flux-framework.org/>`_ "flux" is modern resource manager that can span the space between
+cloud and HPC. If your center does not provide Flux for you, you can `build Flux on your own <https://flux-framework.readthedocs.io/en/latest/quickstart.html#building-the-code>`_
+and launch jobs under your resource manager of choice (e.g., SLURM or on the cloud.)
+
+
+Tutorial
+========
+
+In the `docker/flux <https://github.com/nextflow-io/nextflow/tree/master/docker/flux>`_ directory we provide 
+a `Dockerfile for interacting with Flux <https://github.com/nextflow-io/nextflow/tree/master/docker/flux/.devcontainer/Dockerfile>`_
+
+along with a `VSCode Developer Container <https://code.visualstudio.com/docs/devcontainers/containers>`_
+environment that you can put at the root of the project to be provided with a Flux agent and the dependencies
+needed to build Nextflow. There are two ways to use this:
+
+ - Build a container from scratch and bind your code to it (e.g., for development or testing)
+ - Use VSCode and DevContainers to create a more seamless environment.
+
+Both strategies are described below. For this tutorial, you will generally want 
+to prepare a workflow to use the ``flux`` executor, create an environment with Flux, start a Flux instance, and
+interact with it.
+
+Prepare Nextflow Workflow 
+-------------------------
+
+For your workflow, to use Flux you'll want to specify it in your config. Here is an example 
+``nextflow.config``
+
+
+.. code-block:: console
+
+    // This is an example config for running with flux.
+
+    manifest {
+      mainScript = 'demo.nf'
+      homePage = 'https://github.com/nextflow-io/nextflow/tree/master/docker/flux'
+      description = 'Demo using Nextflow with Flux'
+    } 
+
+    process {
+      executor = 'flux'
+    }
+
+Note that the executor is ``flux``. For additional settings for Flux, see :ref:`flux-executor`.
+Here is an a demo workflow ``demo.nf`` of a job we want to run!
+
+.. code-block:: console
+
+    nextflow.enable.dsl=1
+
+    breakfast = Channel.from '🥞️', '🥑️', '🥧️', '🍵️', '🍞️'
+
+    process sayHello {
+      debug true
+      input:
+      val food from breakfast
+
+      """
+      printf '$food for breakfast!'
+      """
+    }
```

Comment:
```suggestion

    breakfast = Channel.from '🥞️', '🥑️', '🥧️', '🍵️', '🍞️'
 
    workflow {
      sayHello(breakfast)
    }
    
    process sayHello {
      debug true
      input:
      val food

      """
      printf '$food for breakfast!'
      """
    }
```

Love this example 😄. Note DSL1 is deprecated and it's going to be removed in the upcoming version. I've rewrote the it in DSL2

Reply:
okay I see why the files changed - when I do `make compile` that seems to change the versions and timestamps. I was doing that to test the new binary, eg., here is testing the updated workflow:

```bash
$ make compile
# files change
$ nextflow -c nextflow.config run demo.nf
N E X T F L O W  ~  version 22.11.0-SNAPSHOT
Launching `demo.nf` [nice_golick] DSL2 - revision: 58592626a9
executor >  flux (5)
[ae/a01395] process > haveMeal (1) [100%] 5 of 5 ✔
🥑️ for breakfast!
🥧️ for breakfast!
🍞️ for breakfast!
🍵️ for breakfast!
🥞️ for breakfast!
```
I'll try doing `make compile` and then git stash to immediately undo the changes.